plugins/json-schema/src/setup.factory.ts

```diff
@@ -1,81 +1,83 @@
 import { Flatfile } from '@flatfile/api'
-import { Setup, SetupFactory } from '@flatfile/plugin-space-configure'
+import { SetupFactory } from '@flatfile/plugin-space-configure'
 import axios from 'axios'
 
-export interface ModelToSheetConfig extends PartialSheetConfig {
-  sourceUrl: string
+export type JsonSetupFactory = {
+  workbooks: PartialWorkbookConfig[]
+  space?: Partial<Flatfile.spaces.SpaceConfig>
 }
 
 export type PartialWorkbookConfig = Omit<
   Flatfile.CreateWorkbookConfig,
-  'sheets' | 'name'
+  'sheets'
 > & {
-  name?: string
+  sheets: PartialSheetConfig[]
 }
+
 export type PartialSheetConfig = Omit<
   Flatfile.SheetConfig,
   'fields' | 'name'
 > & {
   name?: string
+  source: any | (() => any | Promise<any>)
```

Comment:
Can we make this typing a bit more - strict? I don't think we want to be super strict ajv validation but maybe we can do `object` instead of `any`